tests/s3_test.go

```diff
@@ -0,0 +1,85 @@
+package tests
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/gruntwork-io/terratest/modules/aws"
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/gruntwork-io/terratest/modules/retry"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTerraformS3Bucket(t *testing.T) {
+	t.Parallel()
+
+	expectedName := fmt.Sprintf("terratest-aws-s3-example-%s", strings.ToLower(random.UniqueId()))
+
+	awsRegion := aws.GetRandomStableRegion(t, nil, nil)
+
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		TerraformDir: "../examples/test_minimal",
+
+		Vars: map[string]interface{}{
+			"test_bucket_name": expectedName,
+			"aws_region":       awsRegion,
+		},
+
+		EnvVars: map[string]string{
+			"AWS_DEFAULT_REGION": awsRegion,
+		},
+	})
+
+	defer terraform.Destroy(t, terraformOptions)
+
+	terraform.InitAndApply(t, terraformOptions)
+
+	b, err := terraform.OutputMapOfObjectsE(t, terraformOptions, "test-bucket")
```

Comment:
Try to avoid 1 letter variable names to improve readability.

Reply:
In Go it's pretty common to have 1 letter variables for things that are self-explanatory and used in a small scope. (such as "bucket", "key" in these examples). It's better as well when there is familiarity with patterns, for instance the `t` referring to the `*testing.T` variable, or in S3 context `b` is bucket and `k` is key. I agree though that I could have used best naming conventions to make this less confusing. For example, I used `b` for the output map and `bucket_name` for the bucket name string instead of `b_name`, which would keep the convention that `b` means bucket. 
That being said, I'm open to start using more descriptive names and using less 1 letter variable names if that would make things better for you and your team.